lib/access-context.js

```diff
@@ -193,6 +196,28 @@ AccessContext.prototype.isAuthenticated = function() {
   return !!(this.getUserId() || this.getAppId());
 };
 
+/**
+ * Check if the scope required by the remote method is allowed
+ * by the scopes granted to the requesting access token.
+ * @return {boolean}
+ */
+AccessContext.prototype.isScopeAllowed = function() {
+  if (!this.accessToken) return false;
+
+  // For backwards compatibility, tokens with no scopes are treated
+  // as if they have "DEFAULT" scope granted
+  const scopes = this.accessToken.scopes || ['DEFAULT'];
+
+  // For backwards compatibility, methods with no scopes defined
+  // are assigned a single "DEFAULT" scope
+  const requiredScopes = this.accessScopes || ['DEFAULT'];
+
```

Comment:
I see your point. I find `authorizedScopes` confusing, because it may also mean scopes the access token is authorized to access. I am going to use `resourceScopes` names, inspired by https://github.com/strongloop/loopback/pull/3313#issuecomment-291917954